packages/network-controller/src/NetworkController.ts

```diff
@@ -756,6 +762,69 @@ export class NetworkController extends BaseControllerV2<
     return convertNetworkId(possibleNetworkId);
   }
 
+  /**
+   * Updates the network meta with EIP-1559 support and the network status
+   * based on the given network client ID.
+   *
+   * @param networkClientId - The ID of the network client to update.
+   */
+  async updateNetworkStatus(networkClientId: NetworkClientId) {
```

Comment:
`lookupNetwork` only cares about the currently selected network. the intention of this was to call it when we switch networks, or we could call it automatically. either way we need a new method that doesn't do all the extra things `lookupNetwork` does.

you can check out this ticket for more info: https://github.com/MetaMask/MetaMask-planning/issues/1082

Reply:
for network ID, as far as I know it shouldn't be required for anything and theres been active work to try and get rid of it. Here's a comment from the transaction controller:

![image](https://github.com/MetaMask/core/assets/364566/2ad8dd34-45cf-4622-ba6d-ac92645b4c67)
> Source: https://github.com/MetaMask/core/blob/main/packages/transaction-controller/src/TransactionController.ts#L788-L789

Reply:
Thanks for linking to that ticket and explaining some more context.

We chatted about the network ID, so I'm good with that.

I agree that we need a way to populate the metadata for a network. Does it make sense to have this method be private, at least, and have the network controller call it when the network is switched? Or do we envision this being used multiple ways?

Reply:
I was thinking this could be called in the rpc method for `switchNetwork` when we switch the per-dapp selected network clientId. That or the SelectedNetworkController could have a method that calls it, but the NetworkController wont know about per-dapp selected networks.